src/lib.rs

```diff
@@ -284,6 +284,74 @@ impl<T> ConcurrentQueue<T> {
         }
     }
 
+    /// Attempts to return the value stored at the head of the queue.
+    /// 
+    /// If the queue is empty, an error is returned.
+    /// 
+    /// # Examples
+    /// 
+    /// ```
+    /// use concurrent_queue::{ConcurrentQueue, GetError};
+    /// 
+    /// let mut q = ConcurrentQueue::bounded(2);
+    /// 
+    /// assert_eq!(q.get_head(), Err(GetError::Empty));
+    /// 
+    /// // Push one item and close the queue.
+    /// assert_eq!(q.push(10), Ok(()));
+    /// q.close();
+    /// 
+    /// // Remaining items can be retrieved.
+    /// assert_eq!(q.get_head(), Ok(10));
+    /// // Pop remaining items.
+    /// assert_eq!(q.pop(), Ok(10));
+    /// 
+    /// // Again, get_head errors when the queue is empty,
+    /// // but now also indicates that the queue is closed.
+    /// assert_eq!(q.get_head(), Err(GetError::Closed));
+    /// ```
+    pub fn get_head(&mut self) -> Result<T, GetError> {
```

Comment:
I'm not a fan of the names `get_head` and `get_tail`. How would you feel about `head_mut` and `tail_mut`?

Reply:
Absolutely, I think that looks/sounds better too.